### PR TITLE
Load AequilibraE when opening QGIS project

### LINKED ISSUE #186: [New Feature] Load AequilibraE project on qgis project opening
Now that AequilibraE projects are based on a folder:
QGIS projects can store some python macros (project -> properties -> Macros) to launch for example when opening a qgis project (macro store in qgz file).
We could create a script to load AequilibraE project for the folder in which qgis project is stored (QgsProject.instance().absolutePath())

This way we could store a qgz file in AequilibraE folder and qgis will load AequilibraE project by itself when opening qgz file!

I've tried to create this by inspiring myself with load_project_action.py but cannot achieve it due to the qgis_project parameter that I don't fully understand, can somebody give me a hint ?

### PR DESCRIPTION
**Pull request by r-akemii**
This PR

* Modifies _qaequilibrae.py_ to load previously saved layers and open AequilibraE project from a qgz file;
* Adds _menu_actions/save_as_qgis.py_ file to save temporary layers to an external sqlite database and save the qgis project
* Adds files to initialize save_as_qgis function

TODO:
- [x] Test functions
- [x] Write documentation
- [x] Modify project saving to avoid modifications to project databases when AequilibraE project is closed

### CODE REVIEW

File: qaequilibrae/modules/gis/compare_scenarios_dialog.py [hunk 1]
```diff
@@ -200,8 +200,13 @@ def check_inputs(self):
         return True
 
     def load_result_tables(self):
-        self.link_layer = self.qgis_project.layers["links"][0]
-        QgsProject.instance().addMapLayer(self.link_layer)
+        layer_names = [layer.name() for layer in QgsProject.instance().mapLayers().values()]
```
Comment (pedrocamargo) [inline qaequilibrae/modules/gis/compare_scenarios_dialog.py]: What if there another layer called "links" ???  This is not robust at all

File: qaequilibrae/modules/project_procedures/save_as_qgis.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            self.save_temp_layers_to_db()
+            self.qgz_project.write()
+
+        self.finished.emit("projectSaved")
+
+    def save_temp_layers_to_db(self):
+        for layer in self.qgz_project.mapLayers().values():
+            if layer.isTemporary():
+                layer_name = layer.name() + f"_{uuid4().hex}"
+                print("temp: ", layer.name())
```
Comment (pedrocamargo) [inline qaequilibrae/modules/project_procedures/save_as_qgis.py]: Print???

File: qaequilibrae/modules/public_transport_procedures/transit_navigator_dialog.py [hunk 3]
```diff
@@ -54,20 +54,35 @@ def __init__(self, qgis_project):
         }
 
         fldr = join(dirname(dirname(__file__)), "style_loader")
-        self.stops_layer = qgis_project.layers["transit_stops"][0]
-        self.stops_layer.loadNamedStyle(join(fldr, "stops.qml"), True)
+        layer_names = [layer.name() for layer in QgsProject.instance().mapLayers().values()]
```
Comment (pedrocamargo) [inline qaequilibrae/modules/public_transport_procedures/transit_navigator_dialog.py]: What if the transit layer is NOT the one that comes from the project, but from a different one?  Looking in the list of active layers is a mistake

File: qaequilibrae/modules/routing_procedures/tsp_dialog.py [hunk 4]
```diff
@@ -25,11 +25,19 @@ def __init__(self, qgisproject):
         self.project = qgisproject.project  # type: Project
         self._PQgis = qgisproject
 
-        self.link_layer = self._PQgis.layers["links"][0]
-        self.node_layer = self._PQgis.layers["nodes"][0]
+        layer_names = [layer.name() for layer in QgsProject.instance().mapLayers().values()]
```
Comment (pedrocamargo) [inline qaequilibrae/modules/routing_procedures/tsp_dialog.py]: Again, same thing

File: qaequilibrae/modules/gis/compare_scenarios_dialog.py (hunk 1 above)
Comment (r-akemii) [reply qaequilibrae/modules/gis/compare_scenarios_dialog.py]: I changed the code to check if layers saved in the qgis project are layers from the aequilibrae project, and if so they update the layers dict.

**Comment by pedrocamargo** [comment]
I am tempted to approve this one, but I'd like to see it in action

**Review by pedrocamargo — APPROVED** [review]

**Review by pedrocamargo — APPROVED** [review]

**Review by pedrocamargo — APPROVED** [review]

**Comment by Art-Ev** [comment]
Just a quick message to say that for day-to-day usage, this was a fantastic "upgrade" !

**Pull request merged**